internal/expr/parser.go

```diff
@@ -922,16 +930,36 @@ func (p *Parser) parseColumnsInsertExpr() (expression, bool, error) {
 	}
 	p.skipBlanks()
 
-	parenCol := p.colNum()
-	parenLine := p.lineNum
+	if sources, ok, err := p.parseColumnsInsertValues(); err != nil {
+		cp.restore()
+		return nil, false, err
+	} else if ok {
+		return &columnsInsertExpr{columns: columns, sources: sources, raw: p.input[cp.pos:p.pos]}, true, nil
+	}
+
+	if sources, ok, err := p.parseBasicInsertValues(); err != nil {
```

Comment:
It is a little bit weird that `parseColumnsInsertExpr` has a `parseColumnsInsertValues` and a `parseBasicInsertValues`, it seems like the first one matches the intent and the second one does not belong here naming-wise.

Reply:
Yea, I've just redone this whole section, there was a load of redundancy and confusion there